river/tree/_attribute_observer/numeric_attribute_regression_quantizer_observer.py

```diff
@@ -0,0 +1,163 @@
+import functools
+import math
+import typing
+
+from river.stats import Mean, Var
+from river.utils import VectorDict
+
+from .._attribute_test import AttributeSplitSuggestion, NumericAttributeBinaryTest
+from .attribute_observer import AttributeObserver
+
+
+class NumericAttributeRegressionQuantizerObserver(AttributeObserver):
+    """Quantizer observer (QO).
+
+    Utilizes a dynamical hash-based quantization algorithm to keep track of the target statistics
+    and evaluate split candidates. This class implements the algorithm described in [^1].
+    This attribute observer keeps an internal estimator of the input feature's variance. By doing
+    that, QO can calculate better values for its radius parameter to be passed to future learning
+    nodes.
+
+    Parameters
+    ----------
+    radius
+        The quantization radius.
+
+    References
+    ----------
+    [^1]: Mastelini, S.M. and de Leon Ferreira, A.C.P., 2021. Using dynamical quantization to
+    perform split attempts in online tree regressors. Pattern Recognition Letters.
+
+    """
+
+    def __init__(self, radius: float = 0.01):
+        super().__init__()
+        self.radius = radius if radius > 0 else 0.01
+        self._x_var = Var()
+        self._quantizer = FeatureQuantizer(radius=self.radius)
+
+    def update(self, x, y, sample_weight):
+        if x is None:
+            return
+        else:
+            self._x_var.update(x, sample_weight)
+            self._quantizer.update(x, y, sample_weight)
+
+    def probability_of_attribute_value_given_class(self, x, y):
+        raise NotImplementedError
+
+    def best_evaluated_split_suggestion(
+        self, criterion, pre_split_dist, att_idx, binary_only=True
+    ):
+        candidate = AttributeSplitSuggestion(None, [{}], -math.inf)
+
+        p_x = None
```

Comment:
The meaning of `p_x` is not clear.

Reply:
I renamed it to `prev_x` and added a comment stating that this variable carries the previous value of `x`. Both changes were implemented in commit [fec06b8](https://github.com/online-ml/river/pull/476/commits/fec06b85a8b38f880bc9286c2ef570edc549717f).